src/Microsoft.AspNet.Mvc.Core/ResponseCache.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// An action filter which sets the appropriate headers related to Response caching.
+    /// </summary>
+    public class ResponseCache : ActionFilterAttribute
+    {
+        // A nullable-int cannot be used as an Attribute parameter.
+        // Hence this nullable-int is present to back the Duration property.
+        private int? duration = null;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ResponseCache"/>
+        /// </summary>
+        public ResponseCache()
+        {
+            Location = ResponseCacheLocation.Default;
+            NoStore = false;
+            VaryByHeader = null;
+        }
+
+        /// <summary>
+        /// Duration for which the data from a particular URL must be cached.
+        /// </summary>
+        public int Duration
+        { 
+            get
+            {
+                return (int)duration;
```

Comment:
`return duration ?? -1`

Document that `duration < 0` doesn't set a duration. Really by the time you do all of that, you might as well just not use a nullable. I love me some nullable but it doesn't seem like a great fit here because of attribute restrictions.

What does the old code do?